spec/release-lockfile.spec.json

```diff
@@ -180,6 +177,51 @@
         }
       }
     },
+    "BytecodeObject": {
+      "title": "Contract bytecode",
+      "type": "object",
+      "required": ["bytecode"],
+      "properties": {
+        "bytecode": {
+          "type": "string",
+          "pattern": "^0x[0-9a-fA-F]*$"
+        },
+        "link_references": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/LinkReference"
+          }
+        },
+        "link_dependencies": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/LinkValue"
+          }
+        }
+      }
+    },
+    "LinkReference": {
+      "title": "A defined location in some bytecode which requires linking",
+      "type": "object",
+      "required": [
+        "offset",
+        "length",
+        "name"
+      ],
+      "properties": {
+        "offset": {
+          "type": "integer",
+          "minimum": 0
+        },
+        "length": {
+          "type": "integer",
+          "minimum": 1
+        },
+        "name": {
+          "$ref": "#/definitions/ContractInstanceName"
```

Comment:
I think this could be generalized to a new `Name` or `Identifier` definition - `name` here can be anything, and does not need to refer specifically to a contract instance.